### PR TITLE
Migrate to GitHub Actions

### PR DESCRIPTION
**Pull request by adiroiban**
This removed the Travis-CI code and add GitHub Actions code.

The documentation site was also updated via GitHub Actions

Docs pushed as part of the test at https://twisted.github.io/incremental/docs/  
Check update time.

Check that changes make sense.

I have not used Read The Docs, as the documentation is only pydoctor for now.

Check badges here https://github.com/twisted/incremental/tree/gha-migration

### CODE REVIEW

**Comment by adiroiban** [comment]
I don't know why the links for pypi and coverage are not working for the README badges  https://github.com/twisted/incremental/tree/gha-migration

File: codecov.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        # Temporary allow for a bit of slack in overall code coverage due to
+        # swinging coverage that is not triggered by changes in a PR.
+        # See: https://twistedmatrix.com/trac/ticket/10170
+        threshold: 0.02%
+
+
+# We don't want to receive general PR comments about coverage.
+# We have the commit status checks and that should be enough.
+# See https://docs.codecov.io/docs/pull-request-comments
+comment: false
```
Comment (graingert) [inline codecov.yaml:38]:
```suggestion
comment: true
```

Comment (graingert) [reply codecov.yaml:38]: These are actually very helpful

File: codecov.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        # New code should have 100% CI coverage as the minimum
+        # quality assurance measurement.
+        # If there is a good reason for new code not to have coverage,
+        # add inline pragma comments.
+        target: 100%
+    project:
+      default:
+        # Temporary allow for a bit of slack in overall code coverage due to
+        # swinging coverage that is not triggered by changes in a PR.
+        # See: https://twistedmatrix.com/trac/ticket/10170
```
Comment (graingert) [inline codecov.yaml:31]: Does this apply to incremental?

**Review by graingert — APPROVED** [review]
Can you also add tox-gh-actions ?

**Comment by graingert** [comment]
Merging to unblock other PRs

File: codecov.yaml (hunk 2 above)
Comment (adiroiban) [reply codecov.yaml:31]: I think that this is valid for any project... due to

>  There is also codecov, that will report a decrease in project coverage when a line is removed

File: codecov.yaml (hunk 1 above)
Comment (adiroiban) [reply codecov.yaml:38]: we can enable them for incremental.

**Comment by adiroiban** [comment]
Thanks for the merge. Much appreciated.

> Can you also add tox-gh-actions ?

I am not sure how tox-gh-actions handles the environment variables and how it can be integrated with the GitHub Pages publish.

I am not a fan of tox in general...so I don't want another layer on top of tox for running GitHub actions.
At the same time, I don't feel that the GitHub Action matrix is too complicated to understand so that we would need another tool to manage it.

so... in general, I don't see the value of tox-gh-actions :)

**Pull request merged**